raphtory/src/serialise/parquet/graph.rs

```diff
@@ -21,41 +23,43 @@ pub fn encode_graph_tprop(g: &GraphStorage, path: impl AsRef<Path>) -> Result<()
         1,
         path,
         GRAPH_T_PATH,
-        |_| vec![Field::new(TIME_COL, DataType::Int64, false)],
+        |_| {
+            vec![
+                Field::new(TIME_COL, DataType::Int64, false),
+                Field::new(SECONDARY_INDEX_COL, DataType::UInt64, true),
+            ]
+        },
         |_, g, decoder, writer| {
+            // Each prop key can have multiple values over time.
+            // Flatten into (time, key, value) tuples to group by time.
             let merged_props = g
                 .properties()
                 .temporal()
                 .into_iter()
-                .map(|(k, view)| view.into_iter().map(move |(t, prop)| (k.clone(), t, prop)))
-                .kmerge_by(|(_, t1, _), (_, t2, _)| t1 < t2);
-
-            let mut row = HashMap::<ArcStr, Prop>::new();
-            let mut rows = vec![];
-            let mut last_t: Option<i64> = None;
-            for (key, t1, prop) in merged_props {
-                if let Some(last_t) = last_t {
-                    if last_t != t1 {
-                        let mut old = HashMap::<ArcStr, Prop>::new();
-                        std::mem::swap(&mut row, &mut old);
-                        rows.push(Row {
-                            t: last_t,
-                            row: old,
-                        });
-                    }
-                }
-
-                row.insert(key, prop);
-                last_t = Some(t1);
-            }
-            if !row.is_empty() {
-                rows.push(Row {
-                    t: last_t.unwrap(),
-                    row,
-                });
-            }
+                .map(|(prop_key, prop_view)| {
```

Comment:
I've removed the kmerge in favor of directly grouping by `TimeIndexEntry ` using `chunk_by`. The tests all pass, but lmk if I'm doing something I shouldn't be here.